### DISCUSSION TITLE
Log in via GraphQL query using OIDC provider

### DISCUSSION THREAD

**Discussion by smcintosh80** [requarks/wiki#5480] [Help / Questions] [answered]
I'm trying to design an iOS app for interfacing with Wiki.js. I cannot figure out how to get the OIDC redirect info for logging in via an OIDC provider. When I submit the following query, I get nothing in return:

```
mutation Login {
  authentication {
    login(username: "user@domain.tld", password: "password", strategy: "strategy_key") {
      responseResult {
        succeeded
        errorCode
        slug
        message
      }
      redirect
    }
  }
}
```

**Comment by NGPixel** [comment, marked as answer]
The `login` mutation is strictly for strategies using the user/pass form, not OIDC-type strategies. You would instead query for the `activeStrategies` (see below) to get the `key` for all strategies, then redirect the user to `/login/THE_STRATEGY_KEY_HERE` (replace with the strategy key). The user will be redirected to the OIDC login endpoint automatically.

```graphql
{
  authentication {
    activeStrategies (enabledOnly: true) {
      key
      displayName
    }
  }
}
```

**Reply by smcintosh80** [reply]
Thanks. Not understanding the login flow was keeping me from starting.

I assume this works for all strategies that are not "Local".

**Reply by NGPixel** [reply]
No, only strategies that don't rely on the user/pass form. For example, LDAP relies on the form and wouldn't work.
